File: robot_examples/go/rest_api/swagger/EventExplosion.go
package swagger

import (
)

type EventExplosion struct {
    EventType  Number  `json:"eventType,omitempty"`
    ActivationTime  float32  `json:"activationTime,omitempty"`
    Robot  RobotInfo  `json:"robot,omitempty"`
    HitRobot  RobotInfo  `json:"hitRobot,omitempty"`
    Damage  float32  `json:"damage,omitempty"`
    
}

File: robot_examples/haskell-servant/rest_api/lib/Model/EventExplosion.hs
{-# LANGUAGE DataKinds #-}
{-# LANGUAGE DeriveGeneric #-}
{-# LANGUAGE TypeOperators #-}
{-# LANGUAGE OverloadedStrings #-}
{-# LANGUAGE GeneralizedNewtypeDeriving #-}

module Model.EventExplosion
    ( EventExplosion (..)
    ) where

import Data.Aeson
import GHC.Generics
import Test.QuickCheck
import Model.Event
import Model.Number
import Model.RobotInfo


data EventExplosion = EventExplosion
    { eventType :: Number
    , activationTime :: Float
    , robot :: RobotInfo
    , hitRobot :: RobotInfo
    , damage :: Float
    } deriving (Show, Eq, Generic)

instance FromJSON EventExplosion
instance ToJSON EventExplosion
instance Arbitrary EventExplosion where
    arbitrary = EventExplosion <$> arbitrary <*> arbitrary <*> arbitrary <*> arbitrary <*> arbitrary
